Funky Page

EGideon01 opened the pull request:
Heavily based upon the original FlexBox demo in Bryce.html, this includes a new Funky.html page that provides an assortment of flex boxes. These boxes currently contain the same functionality as before, but goal for tomorrow or Saturday is get the boxes to rotate when a singular button is clicked. Hoping this function can be performed using FlexBox.

EGideon01 committed:
Funky Files

This includes a new Funky page that continues to test and demo the flexbox and flex directions. Goal tomorrow or Saturday is to get the flexboxes to rotate in a cascading pattern. Hoping there's an area of FlexBox to allow for such. Also doing an additional pr for the relative changes to the other files to account for the new Funky page.

EGideon01 committed:
Includes New Funky Button

EGideon01 committed:
Includes New Funky Button

EGideon01 committed:
Includes New Funky Button

EGideon01 committed:
Includes New Funky Button

brycecrist commented on Funky.css:
```diff
@@ -0,0 +1,73 @@
+    width: 8%;
+    height: 3rem;
+    box-shadow: 4px 4px 0px 0px rgba(0,0,0,0.65);
+}
```
Consider white space :)

brycecrist commented on Funky.css:
```diff
@@ -0,0 +1,73 @@
+    height: 5rem;
+    box-shadow: 4px 4px 0px 0px rgba(0,0,0,0.65);
+}
+#FunkyArea {
+    display: flex;
+    justify-content: center;
+    flex-direction: row;
+    width: 100%;
+    margin: 1em;
+}
+#FunkyArea2 {
+    display: flex;
+    justify-content: center;
+    flex-direction: row;
+    width: 100%;
+    margin: 1em;
+}
```
IIRC stuff like this can be written:
```
#FunkyArea, #FunkyArea2 {
    display: flex;
    justify-content: center;
    flex-direction: row;
    width: 100%;
    margin: 1em;
}
```

Also just FYI while it's fine here -- naming stuff like "Thing1" and "Thing2" is generally frowned upon

brycecrist replied:
But you could consolidate a lot of this CSS. Also this is a good use case for CSS classes as well. Instead of giving ids like 1, 2, 3, 4, 5, and so on, make each area the same class, thus they should have the same CSS.
```
.FunkyArea {

}
```
^ This will target every element with that classname. Alternatively, if each element is the same element e.g. `div` you could target the parent:
```
#ParentOfFunkyPage div {

}
```
^ Targets all divs under the parent.

If you find yourself copy-pasting, take a step back and make sure it's needed.

brycecrist commented on Funky.html:
```diff
@@ -0,0 +1,68 @@
+        <div class="funky">4</div>
+        <div class="funky">5</div>
+    </section>
+    <section id="FunkyArea2">
```
Yep each of these sections could just be `class="FunkyArea"` unless you need to specifically target them via JS. And even then, you still could: `var funkyAreas = document.getElementsByClassName("FunkyArea")` Then you could target the 4th instance by: `var funkyArea4 = funkyAreas[3]` <- accessing the their index.

This is a really "repeat yourself" way of doing things. We like to keep things DRY.

Also a great usecase for React, but I won't go there ;)

brycecrist commented on Funky.js:
```diff
@@ -0,0 +1,50 @@
+    else if (currentDirection === "column")
+        flexBox.style.flexDirection = "row"
+}
+const funkyRotate2 = () => {
```
lmao yeah you're copy pasting code. Big bad. You should have a single method that does this, where you pass in the element you want. This sort of code would always be denied in PR reviews

EGideon01 committed:
Update Funky.html

OK! It is ~midnight and I am still stuck on trying to get the "const funkyRotate" function to actually call on the right aspect of the new funkyList var. Gonna be committing the new Java and CSS with this to show how rough I made this. Currently, selecting a button only moves the first row, instead of the row related to the button. With that in mind, I am unsure on how to actually point to a specific section in the funkyList array for the flexBox const (which, in turn, leads to the function just always defaulting to the first row, instead of the row indicated by the array #). Plz halp );

EGideon01 committed:
Update Funky.js

This is the js responsible for site no worky ):

EGideon01 committed:
Update Funky.css

On brighter news for this set of commits, this css includes all new cool styling for the funky page. Pretty happy with the way this page came out. (:
